activerecord/lib/active_record/insert_all.rb

```diff
@@ -37,7 +42,7 @@ def update_duplicates?
 
     def map_key_with_value
       inserts.map do |attributes|
-        attributes = attributes.stringify_keys
+        attributes = scope_for_create.reverse_merge(attributes.stringify_keys)
```

Comment:
Can just be `attributes.stringify_keys.merge(scope_for_create)` instead of `reverse_merge` — it's a little more straightforward that way.